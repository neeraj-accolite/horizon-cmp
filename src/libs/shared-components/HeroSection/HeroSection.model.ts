export interface HeroSectionProps {
  /**
   * The headline text of the page.
   */
  title: string;

  /**
   * High level overview of specific group of services
   */
  overview: string;

  /**
   * Prominent inline callout in hero section if specific service is only offered in certain geographic areas and geolocation-based experience is not feasible
   */
  calloutText: string;

  /**
   * CTA button text
   */
  ctaButtonText: string;

  /**
   * The click event listener on the CTA button clicked of the Hero component.
   */
  onHeroCtaClicked: () => void;
}
