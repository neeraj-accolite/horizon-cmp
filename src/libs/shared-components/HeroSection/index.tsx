export { default as HeroSection } from './HeroSection';

export type { HeroSectionProps } from './HeroSection.model';
