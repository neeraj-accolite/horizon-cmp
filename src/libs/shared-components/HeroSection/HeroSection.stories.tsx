import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import HeroSection from './HeroSection';
import { HeroSectionProps } from './HeroSection.model';
export default {
  title: 'Components/HeroSection',
  component: HeroSection,
} as Meta;

const Template: StoryFn<HeroSectionProps> = (args) => <HeroSection {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'Plumbing',
  overview:
    'Whether your problem feels like a drop in the bucket or a giant puddle of woes, Horizon is here to help. <br /> <br />Our highly trained plumbers handle it all, from minor leak repairs to full system replacement and installation.',
  calloutText: 'Inline callout if service is only offered in specific areas',
  ctaButtonText: 'Book Online',
  onHeroCtaClicked: () => {
    alert('The Book online button is clicked');
  },
};
