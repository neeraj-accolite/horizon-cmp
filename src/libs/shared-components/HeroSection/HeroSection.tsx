'use client';

import Image from 'next/image';
import { Button } from '../Button';
import { HeroSectionProps } from './HeroSection.model';

const HeroSection = ({
  title,
  overview,
  calloutText,
  ctaButtonText,
  onHeroCtaClicked,
}: HeroSectionProps) => {
  return (
    <div className="mx-auto grid grid-cols-1 justify-between bg-hero-background px-4 py-10 sm:grid-cols-2 sm:px-24 sm:py-14">
      <div className="content-center justify-center sm:mr-28">
        <h1 className="text-center text-hero-heading sm:text-left">{title}</h1>
        <div className="whitespace-pre-wrap py-4 text-hero-overview sm:py-6">
          {/**
           * TODO: Change the inner html implementation based on the confirmation on
           * getting text/html from Drupal backend.
           */}
          <div dangerouslySetInnerHTML={{ __html: overview }} />
        </div>
        <div className="mb-2 pb-4">
          <span className="font-bold text-hero-callout">{calloutText}</span>
        </div>
        <div className="mt-2">
          <Button
            className="sm:!max-w-xs"
            color="primary"
            onClick={onHeroCtaClicked}
          >
            {ctaButtonText}
          </Button>
        </div>
      </div>
      <div className="mt-2 flex content-center justify-center sm:mt-0">
        {/**
         * TODO: This Review image Component will be replaced later with actual component in next story.
         * Currently, the image is used to design the layout of the Hero component.
         */}
        <div className="content-center">
          <Image
            src={'/Review.png'}
            width={280}
            height={72}
            alt={'review component'}
          />
        </div>
      </div>
    </div>
  );
};

export default HeroSection;
