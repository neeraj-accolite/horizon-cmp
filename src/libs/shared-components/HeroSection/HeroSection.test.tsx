import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { HeroSectionProps } from './HeroSection.model';
import HeroSection from './HeroSection';

describe('UI Component- Hero Section', () => {
  const defaultProps: HeroSectionProps = {
    title: 'Plumbing',
    overview:
      'Whether your problem feels like a drop in the bucket or a giant puddle of woes, Horizon is here to help. <br /> <br />Our highly trained plumbers handle it all, from minor leak repairs to full system replacement and installation.',
    calloutText: 'Inline callout if service is only offered in specific areas',
    ctaButtonText: 'Book Online',
    onHeroCtaClicked: jest.fn(),
  };

  test('Hero Section title rendered successfully with H1 heading text', () => {
    render(<HeroSection {...defaultProps} />);
    const titleElement = screen.getByRole('heading', {
      level: 1,
    });
    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveTextContent(defaultProps.title);
  });

  test('Hero Section CTA button callback is clickable ', () => {
    const container = render(<HeroSection {...defaultProps} />);
    const ctaButton = container.getByText(defaultProps.ctaButtonText);
    expect(ctaButton).toBeInTheDocument();
    fireEvent.click(ctaButton);
    expect(defaultProps.onHeroCtaClicked).toHaveBeenCalled();
  });

  test('The snapshot is matches of the UI component', () => {
    const { container } = render(<HeroSection {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
