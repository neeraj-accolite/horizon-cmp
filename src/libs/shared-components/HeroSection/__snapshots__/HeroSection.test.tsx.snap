// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component- Hero Section The snapshot is matches of the UI component 1`] = `
<div>
  <div
    class="mx-auto grid grid-cols-1 justify-between bg-hero-background px-4 py-10 sm:grid-cols-2 sm:px-24 sm:py-14"
  >
    <div
      class="content-center justify-center sm:mr-28"
    >
      <h1
        class="text-center text-hero-heading sm:text-left"
      >
        Plumbing
      </h1>
      <div
        class="whitespace-pre-wrap py-4 text-hero-overview sm:py-6"
      >
        <div>
          Whether your problem feels like a drop in the bucket or a giant puddle of woes, Horizon is here to help. 
          <br />
           
          <br />
          Our highly trained plumbers handle it all, from minor leak repairs to full system replacement and installation.
        </div>
      </div>
      <div
        class="mb-2 pb-4"
      >
        <span
          class="font-bold text-hero-callout"
        >
          Inline callout if service is only offered in specific areas
        </span>
      </div>
      <div
        class="mt-2"
      >
        <a
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover button primary sm:!max-w-xs"
          role="button"
          tabindex="0"
        >
          Book Online
        </a>
      </div>
    </div>
    <div
      class="mt-2 flex content-center justify-center sm:mt-0"
    >
      <div
        class="content-center"
      >
        <img
          alt="review component"
          data-nimg="1"
          decoding="async"
          height="72"
          loading="lazy"
          src="/_next/image?url=%2FReview.png&w=640&q=75"
          srcset="/_next/image?url=%2FReview.png&w=384&q=75 1x, /_next/image?url=%2FReview.png&w=640&q=75 2x"
          style="color: transparent;"
          width="280"
        />
      </div>
    </div>
  </div>
</div>
`;
